src/compiler/checker.ts

```diff
@@ -1473,8 +1476,6 @@ namespace ts {
             return undefined;
         }
 
-        // This is for caching the result of getSymbolDisplayBuilder. Do not access directly.
-        let _displayBuilder: SymbolDisplayBuilder;

```

Comment:
It was. It was unreachable code - if this was ever compiled to a `let` in the result rather than a `var`, the resulting TS wouldn't have run correctly. :D

`let` declarations after a return statement are unreachable and are therefore never scoped, unlike a var which would be hoisted with the function definition.

So you can say the linter found a bug waiting to happen here.